org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/AnnotationScanTest.java

```diff
@@ -258,7 +258,7 @@ public void testInjectedContext() throws Exception {
 	}
 
 	private IRunnableDevice<ScanModel> createGridScan(IRunnableDevice<?> detector,
-			IScannable<?> monitor, int... size) throws Exception {
+			IScannable<?> monitorsPerPoint, IScannable<?> monitorsPerScan, int... size) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)